File: app/src/main/java/com/ab/ble/view/SettingsActivity.java
package com.ab.ble.view;

import android.arch.lifecycle.LifecycleRegistry;
import android.arch.lifecycle.LifecycleRegistryOwner;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.RadioButton;

import com.ab.ble.R;
import com.ab.ble.veiwmodel.SettingsViewModel;

public class SettingsActivity extends AppCompatActivity implements LifecycleRegistryOwner {

    private final LifecycleRegistry lifecycleRegistry = new LifecycleRegistry(this);

    private SettingsViewModel mViewModel;

    private RadioButton mContinuousRB;
    private RadioButton mManualRB;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        mContinuousRB = findViewById(R.id.continuous_rb);
        mManualRB = findViewById(R.id.manual_rb);

        mViewModel = ViewModelProviders.of(this).get(SettingsViewModel.class);
        mViewModel.getIsContinuousScan().observe(this, this::updateDataAndUI);
    }

    @Override
    public LifecycleRegistry getLifecycle() {
        return lifecycleRegistry;
    }

    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.continuous_layout:
                mViewModel.onContinuousLayoutClicked();
                break;
            case R.id.manual_layout:
                mViewModel.onManualLayoutClicked();
                break;
        }
    }

    private void updateDataAndUI(boolean isContinuous) {
        mContinuousRB.setChecked(isContinuous);
        mManualRB.setChecked(!isContinuous);
    }
}

File: app/src/main/java/com/ab/ble/view/SettingsActivity.kt
package com.ab.ble.view

import android.arch.lifecycle.LifecycleRegistry
import android.arch.lifecycle.LifecycleRegistryOwner
import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.view.View
import com.ab.ble.R
import com.ab.ble.veiwmodel.SettingsViewModel
import kotlinx.android.synthetic.main.activity_settings.*

class SettingsActivity : BaseHomeUpActivity(), LifecycleRegistryOwner {

    private val lifecycleRegistry = LifecycleRegistry(this)

    private var mViewModel: SettingsViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        mViewModel = ViewModelProviders.of(this).get(SettingsViewModel::class.java)
        mViewModel!!.isContinuousScanLD.observe(this, Observer<Boolean> { isContinuous -> updateDataAndUI(isContinuous!!) })
        mViewModel!!.scanTimeLD.observe(this, Observer<Int> { it -> onScanTimeChange(it!!) })
    }

    override fun getLifecycle(): LifecycleRegistry {
        return lifecycleRegistry
    }

    fun onClick(v: View) {
        when (v.id) {
            R.id.continuous_layout -> mViewModel!!.onContinuousLayoutClicked()
            R.id.manual_layout -> {
                mViewModel!!.onManualLayoutClicked()
                NumberPickerDialogFragment.showFragment(supportFragmentManager)
            }
        }
    }

    private fun updateDataAndUI(isContinuous: Boolean) {
        continuous_rb!!.isChecked = isContinuous
        manual_rb!!.isChecked = !isContinuous
    }

    private fun onScanTimeChange(integer: Int) {
        scan_time!!.text = getString(R.string.scan_time_label, integer)
    }

    companion object {
        private val TAG = SettingsActivity::class.java.simpleName
    }
}